### PR TITLE
Update build_devctr with hint about version choice

### PR DESCRIPTION
**Pull request by roypat**
Signed-off-by: Patrick Roy <roypat@amazon.co.uk>

## Changes

Add instructions to make sure that we do not accidentally overwrite existing images when publishing new versions of the devctr

## Reason

We accidentally overwrote some images the other day

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by luminitavoicu — DISMISSED** [review]

File: docs/devctr-image.md [hunk 1]
```diff
@@ -80,7 +80,10 @@ registry. The Firecracker CI suite must also be updated to use the new image.
     ```
 
 1. Tag the new image with the next available version and the architecture
-   you're on.
+   you're on. Note that this will not always be "current version in devtool + 1",
+   as sometimes that version might already be used on feature branches. Always
+   check the "Image Tags" on [the fcuvm repository](https://gallery.ecr.aws/firecracker/fcuvm)
+   to make sure you do not accidentally overwrite existing image.
```
Comment (pb8o) [inline docs/devctr-image.md:87]: Maybe can add this as a sanity check before push i.e. this should fail: 

```sh
docker pull public.ecr.aws/firecracker/fcuvm:v47

Error response from daemon: manifest for public.ecr.aws/firecracker/fcuvm:v47 not found: manifest unknown: Requested image not found
```

**Review by pb8o — APPROVED** [review]

File: docs/devctr-image.md (hunk 1 above)
Comment (roypat) [reply docs/devctr-image.md:87]: Good point, added it!

**Review by luminitavoicu — DISMISSED** [review]

**Review by pb8o — APPROVED** [review]

File: docs/devctr-image.md [hunk 2]
```diff
@@ -79,8 +79,22 @@ registry. The Firecracker CI suite must also be updated to use the new image.
     public.ecr.aws/firecracker/fcuvm   v26       8d00deb17f7a     2 weeks ago   2.41GB
     ```
 
-1. Tag the new image with the next available version and the architecture
-   you're on.
+1. Tag the new image with the next available version `X` and the architecture
+   you're on. Note that this will not always be "current version in devtool + 1",
+   as sometimes that version might already be used on feature branches. Always
+   check the "Image Tags" on [the fcuvm repository](https://gallery.ecr.aws/firecracker/fcuvm)
+   to make sure you do not accidentally overwrite existing image.
```
Comment (JonathanWoollett-Light) [inline docs/devctr-image.md:86]: Nit
```suggestion
   to make sure you do not accidentally overwrite an existing image.
```

File: docs/devctr-image.md [hunk 3]
```diff
@@ -79,8 +79,22 @@ registry. The Firecracker CI suite must also be updated to use the new image.
...
+
+   As a sanity check, run
```
Comment (JonathanWoollett-Light) [inline docs/devctr-image.md:88]: Nit
```suggestion
   As a sanity check, run:
```

**Review by JonathanWoollett-Light — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Pull request merged**